Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m

```diff
@@ -42,22 +40,25 @@ - (instancetype)initWithTox:(OCTTox *)tox
     _audioEngine.toxav = self.toxAV;
     [_audioEngine setupWithError:nil];
 
-    OCTConverterFriend *friendConverter = [OCTConverterFriend new];
-    friendConverter.dataSource = self;
-
-    OCTConverterMessage *messageConverter = [OCTConverterMessage new];
-    messageConverter.converterFriend = friendConverter;
-
-    _chatConverter = [OCTConverterChat new];
-    _chatConverter.delegate = self;
-    _chatConverter.converterFriend = friendConverter;
-    _chatConverter.converterMessage = messageConverter;
-
-    _calls = [OCTCallsContainer new];
-
     return self;
 }
 
+- (BOOL)setupWithError:(NSError **)error
+{
+    NSAssert(self.delegate, @"dataSource is needed before setup of OCTSubmanagerCalls");
+    __block BOOL status = NO;
+    dispatch_once(&_setupOnceToken, ^{
+        OCTRealmManager *realmManager = [self.dataSource managerGetRealmManager];
+        _timer = [[OCTCallTimer alloc] initWithRealmManager:realmManager];
+
+        _audioEngine = [OCTAudioEngine new];
+        _audioEngine.toxav = self.toxAV;
+        status = [_audioEngine setupWithError:error];
```

Comment:
You can use properties instead of ivars here. Generally speaking ivars are only for init method and getters/setters.